client/src/components/Map/UCIMap.js

```diff
@@ -45,6 +48,170 @@ export default class UCIMap extends PureComponent {
+                    const id = locations[buildingCode];
+                    const locationData = buildingCatalogue[id];
+
+                    if (locationData === undefined) return;
+
+                    colors.push(event.color);
+
+                    if (day) {
+                        if (coords) {
+                            coords += ';';
+                        }
+                        coords += locationData.lng + ',' + locationData.lat;
+                        coords_array.push([locationData.lat, locationData.lng]);
+                    }
+                    index++;
+                });
+            if (index > 1) {
+                var url = new URL(DIRECTIONS_ENDPOINT + encodeURIComponent(coords));
+
+                url.search = new URLSearchParams({
+                    alternatives: false,
+                    geometries: 'geojson',
+                    steps: false,
+                    access_token: ACCESS_TOKEN,
+                }).toString();
+
+                fetch(url, {
+                    method: 'GET',
+                    headers: { 'Content-Type': 'application/json' },
+                }).then((response) => {
+                    response.json().then((obj) => {
+                        let coordinates = obj['routes'][0]['geometry']['coordinates']; // The coordinates for the lines of the routes
+                        let waypoints = obj['waypoints']; // The waypoints we specified in the request
+                        let waypointIndex = 0; // The current waypoint we are building a path from
+                        let path = [
+                            [[waypoints[waypointIndex]['location'][1], waypoints[waypointIndex]['location'][0]]],
+                        ]; // Path is a list of paths for each waypoint. For example, path[0] is the path to waypoint 0, path[1] is the path from 0 to 1... etc.
+
+                        let poly = []; // Arrays of polyline to be added to map
+                        let info_markers = [];
+
+                        poly.push(
+                            <Polyline color={colors[0]} positions={[path[0][0], coords_array[0]]} dashArray="4" />
+                        ); // Draw a dashline from waypoint 0 to start of route
+
+                        for (let [lat, lng] of coordinates) {
+                            path[waypointIndex].push([lng, lat]); // Creates a path using lat and lng of coordinates until lat and lng matches one of the waypoint's coordinates
+                            if (
+                                lat === waypoints[waypointIndex]['location'][0] &&
+                                lng === waypoints[waypointIndex]['location'][1]
+                            ) {
+                                path.push([[lng, lat]]);
+                                if (waypointIndex !== 0) {
+                                    poly.push(
+                                        <Polyline
```

Comment:
Add a `zIndexOffset` here, so the info is displayed in front of the pin
| Before | After |
|--|--|
| ![image](https://user-images.githubusercontent.com/29494270/127456490-7ed56f49-1c71-441b-8466-d7989dffc4a4.png) | ![image](https://user-images.githubusercontent.com/29494270/127456652-8d37798a-3d9c-4e80-bca7-3a446e97505f.png) |

```suggestion
                                        <Polyline
                                            zIndexOffset={100}
```